lib/index.d.ts

```diff
@@ -0,0 +1,98 @@
+/// <reference types="typescript" />
+///
+/// https://www.typescriptlang.org/docs/handbook/modules.html
+/// When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.
```

Comment:
I could only get `OktaAuth` be visible but not others. I think it's probably because we're exporting module in cjs style.
Probably OK for first round and look for any suggestions/ideas.

Reply:
I'm wondering if we should release this with partial support now, or wait until we can dedicate further time (with tests) to ensure 100% type coverage.

Once we release support for this, we're going to need to maintain it. 😄

Reply:
Is there any movement on this?  Typescript types would be super helpful.